### PR TITLE
Job validation

### PR DESCRIPTION
**Pull request by nspc97**
https://trello.com/c/stFxfgZr/100-be-jobs-for-validation-list

### CODE REVIEW

**Review by MihaiDruta — APPROVED** [review]

File: src/Controller/ActivityController.php [hunk 1]
```diff
@@ -275,9 +275,11 @@ public function getActivityDetails(Activity $activity): Response
     {
         $user = $this->getUser();
         $rights = $this->accessRightsPolicy->canAccessActivity($activity, $user);
-        $hasAccess = $this->isGranted('ROLE_ADMIN');
+        $hasAccess = $this->isGranted('ROLE_PM');
 
-        if ($rights === false && !$hasAccess) {
+        if (($rights === false && !$hasAccess) ||
+            ($activity->getStatus() === Activity::STATUS_IN_VALIDATION &&
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: A PM has always access to the jobs posted by users for whom they are a PM. Also here you don't check that the current user is a PM of the owner. Also you have remove admin from always having access :)

File: src/Controller/ActivityController.php [hunk 2]
```diff
@@ -470,6 +474,24 @@ public function createAction(
 
         $activityRepository->save($newActivity);
 
+        $message = (new Swift_Message(
+            $owner->getName() . ' ' . $owner->getSurname() .
+            ' has created a new activity waiting for your validation '
+        ))
+            ->setFrom($_ENV['EMAIL_FROM'])
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: Inject the parameter `EMAIL_FROM` and make this change everywhere.

File: src/Controller/ActivityController.php [hunk 3]
```diff
@@ -470,6 +474,24 @@ public function createAction(
...
+            ->setTo($owner->getProjectManager()->getEmail())
+            ->setBody(
+                $this->renderView(
+                    'mail/mail_for_PM.html.twig',
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: `mail_for_PM` doesn't indicate what mail is this. Is it that a user has been assigned to the PM, is it a new job to validate e.t.c.

File: src/Controller/ActivityController.php [hunk 4]
```diff
@@ -470,6 +474,24 @@ public function createAction(
 
         $activityRepository->save($newActivity);
 
+        $message = (new Swift_Message(
```
Comment (SpartakusMd) [inline src/Controller/ActivityController.php]: Create a separate service for sending emails.

File: src/Repository/ActivityRepository.php [hunk 5]
```diff
@@ -149,4 +155,23 @@ public function getPaginatedActivities(
...
+    public function getActivitiesForValidation(
+        User $user
+    ): QueryBuilder {
+        $queryBuilder = $this->createQueryBuilder('activity');
+        $queryBuilder
+            ->select('activity')
+            ->join('activity.owner', 'user')
+            ->where('activity.status = :in_validation')
+            ->setParameter(':in_validation', Activity::STATUS_IN_VALIDATION);
+        $hasAccess = $this->checker->isGranted('ROLE_ADMIN');
```
Comment (SpartakusMd) [inline src/Repository/ActivityRepository.php]: Everywhere the variable name is `$hasAccess`! Why not use something more explicit like `$isAdmin`.

File: src/Transformer/ActivityTransformer.php [hunk 6]
```diff
@@ -164,7 +164,11 @@ public function editTransform(
         $activity->setDescription($dto->description);
         $activity->setApplicationDeadline($dto->applicationDeadline);
         $activity->setFinalDeadline($dto->finalDeadline);
-        $activity->setStatus($dto->status);
+
+        if ($activity->getStatus() !== Activity::STATUS_IN_VALIDATION) {
```
Comment (SpartakusMd) [inline src/Transformer/ActivityTransformer.php:168]: 👍 Very good :)

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

File: src/Service/EmailSender.php [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+use App\Entity\Activity;
+use App\Entity\User;
+use Psr\Container\ContainerInterface;
+use Swift_Mailer;
+use Swift_Message;
+use Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait;
+
+class EmailSender
+{
+    use ControllerTrait;
```
Comment (SpartakusMd) [inline src/Service/EmailSender.php]: ControllerTrait should be used only in Controllers.

**Review by SpartakusMd — CHANGES_REQUESTED** [review]

**Pull request merged**